lib/services/local/saml_idp_service_provider.go

```diff
@@ -183,3 +237,60 @@ func validateSAMLIdPServiceProvider(sp types.SAMLIdPServiceProvider) error {
 
 	return nil
 }
+
+// fetchAndSetEntityDescriptor fetches Service Provider entity descriptor (aka SP metadata)
+// from remote metadata endpoint (Entity ID) and sets it to sp if the xml format
+// is a valid Service Provider metadata format.
+func (s *SAMLIdPServiceProviderService) fetchAndSetEntityDescriptor(sp types.SAMLIdPServiceProvider) error {
+	if s.httpClient == nil {
+		return trace.BadParameter("missing http client")
+	}
+	resp, err := s.httpClient.Get(sp.GetEntityID())
+	if err != nil {
+		return trace.Wrap(err)
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode != http.StatusOK {
+		return trace.Wrap(trace.ReadError(resp.StatusCode, nil))
```

Comment:
Should we read response body and include it in the error and/or log it for easier troubleshooting?